tools/alerts/alert_utils.py

```diff
@@ -0,0 +1,168 @@
+import re
+from typing import Dict, Any, List, Union
+
+import urllib
+import os
+import requests
+import json
+
+ALL_SKIPPED_THRESHOLD = 100
+SIMILARITY_THRESHOLD = 0.75
+FAILURE_CHAIN_THRESHOLD = 2
+MAX_CONCURRENT_ALERTS = 1
+FAILED_JOB_PATTERN = (
+    r"^- \[(.*)\]\(.*\) failed consecutively starting with commit \[.*\]\(.*\)$"
+)
+
+PENDING = "pending"
```

Comment:
Are these duplications from `alert_registry.py` (or may be the other way around)?